email/add_ecr/email_sender.py

```diff
@@ -0,0 +1,311 @@
+'''This script sends a report email to the users of the Bookworm dashboard.'''
+import os
+from dotenv import load_dotenv
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from datetime import datetime
+import boto3
+from botocore.exceptions import ClientError
+import pandas as pd
+import psycopg2
+
+
+def get_db_connection() -> psycopg2.extensions.connection:
+    '''Gets a connection to the DB specified in the .env'''
+    load_dotenv()
+    return psycopg2.connect(
+        database=os.getenv('DB_NAME'),
+        user=os.getenv('DB_USERNAME'),
+        host=os.getenv('DB_HOST'),
+        password=os.getenv('DB_PASSWORD'),
+        port=os.getenv('DB_PORT')
+    )
+
+
+def get_publishers_tracked_authors(publisher_id: int) -> list[tuple]:
+    '''Gets all publishers  and their tracked authors from the DB'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+            SELECT 
+            a.author_id
+            FROM publisher as p
+            LEFT JOIN author_assignment AS aa ON aa.publisher_id = p.publisher_id
+            LEFT JOIN author AS a ON a.author_id = aa.author_id
+            where p.publisher_id = {publisher_id}
+            ;
+            '''
+        cur.execute(sql)
+        return cur.fetchall()
+    finally:
+        conn.close()
+
+
+def get_publisher_ids() -> list[tuple]:
+    '''Gets the ids of all publishers'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+            SELECT 
+            publisher_id
+            FROM publisher
+            ;
+            '''
+        cur.execute(sql)
+        return cur.fetchall()
+    finally:
+        conn.close()
+
+
+def get_publishers_name(publisher_id: int) -> str:
+    '''Returns the name for the specified publisher id'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+            SELECT 
+            publisher_name
+            FROM publisher
+            WHERE publisher_id = {publisher_id}
+            ;
+            '''
+        cur.execute(sql)
+        return cur.fetchone()[0]
+    finally:
+        conn.close()
+
+
+def get_publishers_email(publisher_id: int) -> str:
+    '''Returns the email for the specified publisher id'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+            SELECT 
+            publisher_email
+            FROM publisher
+            WHERE publisher_id = {publisher_id}
+            ;
+            '''
+        cur.execute(sql)
+        return cur.fetchone()[0]
+    finally:
+        conn.close()
+
+
+def get_avg_rating_difference_since_yesterday(author_id) -> int:
```

Comment:
Missing type hint here for `author_id` 
 ```suggestion
def get_avg_rating_difference_since_yesterday(author_id: int) -> int:
```

Reply:
thanks, ill commit this locally instead of on gh.